src/main/java/gregtech/api/util/GTUtility.java

```diff
@@ -249,6 +249,41 @@ public static boolean mergeItemStack(ItemStack itemStack, List<Slot> slots, bool
         return merged;
     }
 
+    /**
+     * Attempts to merge given ItemStack with ItemStacks in list supplied
+     * growing up to their max stack size
+     *
+     * @param stackToAdd item stack to merge.
+     * @return a list of stacks, with optimized stack sizes
+     */
+
+    public static List<ItemStack> addStackToItemStackList(ItemStack stackToAdd, List<ItemStack> itemStackList){
+        if (!itemStackList.isEmpty()) {
+            for (ItemStack stackInList : itemStackList) {
+                if (ItemStackHashStrategy.comparingAllButCount().equals(stackInList, stackToAdd)) {
+                    if (stackInList.getCount() < stackInList.getMaxStackSize()) {
+                        int insertable = stackInList.getMaxStackSize() - stackInList.getCount();
+                        if (insertable >= stackToAdd.getCount()) {
+                            stackInList.grow(stackToAdd.getCount());
+                            stackToAdd = ItemStack.EMPTY;
+                        } else {
+                            stackInList.grow(insertable);
+                            stackToAdd = stackToAdd.copy();
+                            stackToAdd.setCount(stackToAdd.getCount() - insertable);
+                        }
+                        break;
```

Comment:
Why do we `break` here? We would not want to break if `stackToAdd` was not empty, because then there would not be a chance to merge `stackToAdd` into the rest of the list. We should only break if `stackToAdd` gets completely emptied into the list of items.